neurogym/utils/functions.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime, timezone
+from pathlib import Path
+
+
+def ensure_dir(path: Path | str) -> Path:
+    """A thin shim over the Path class.
+
+    Creates a directory if it doesn't exist and
+    returns the resolved and expanded path.
+
+    Args:
+        path (Path | str):
+            A directory as a path or a string.
+
+    Returns:
+        Path:
+            The resolved and expanded path.
+    """
+    path = Path(path)
+    path.mkdir(exist_ok=True, parents=True)
+    return path.expanduser().resolve().absolute()
+
+
+def iso_datetime(ms: bool = False) -> str:
+    """Create a date-timestamp as an ISO-formatted string.
+
+    Useful for adding a unique but meaningful string to the
+    name of a directory or a file that might be created
+    repeatedly with the same name (for instance, when
+    running the same experiment multiple times).
+
+    Args:
+        ms (bool, optional):
+            Use millisecond precision. Defaults to False.
+
+    Returns:
+        str:
+            The formatted timestamp.
```

Comment:
It would be great if you could format the docstrings as follows. 

I realize that the codebase doesn't have perfectly consistent docstring formatting, but at least for the part that we have been updating, we try to adhere to the `function_with_pep484_type_annotations` standard from [google docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). 

```suggestion
    Args:
        ms: Use millisecond precision. Defaults to False.

    Returns:
        str: The formatted timestamp.
```

Note that it the docstring doesn't need the type or the "optional" status, as this follows from the type hint and stating what it defaults to.

Reply:
I'll reformat these (I use `autoDocstring`, which adds the types automatically), but what do you use in existing modules that don't have type hints? Just add them?

Reply:
This is also a general comment on docstrings: I noticed that a lot of classes have docstrings at the class level that really should be in the `__init__()` method. Is there a plan to move those? I can do that gradually, starting with the files that I've changed here (e.g., the `Monitor` class).